### PR TITLE
B #3177 Load max_nbds from running module

### LINKED ISSUE #3177: LXD drivers should load max nbd devices from Linux config
**Description**
The qcow2 mapper [has hard coded 256 max nbd devices ](https://github.com/OpenNebula/one/blob/d8541c1239fe65458394e0589795127262d0e95c/src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb#L25-L27). The config should be loaded from Linux.

**Use case**
Being able to change the config in a particular node without having to change the script later on in the code.

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

### PR DESCRIPTION
**Pull request by dann1**
Use modprobe -c to load max number of nbd devices

Closes #3177

### CODE REVIEW

**Comment by dann1** [comment]
I think this could be added for 5.8.1

File: src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb [hunk 1]
```diff
@@ -67,6 +65,21 @@ def do_unmap(device, _one_vm, _disk, _directory)
...
     private
 
+    # Detects Max number of block devices
+    def load_nbds_max
+        cmd = '/sbin/modprobe -c'
+        rc, mods_conf, e = Command.execute(cmd, false)
+
+        if !rc.zero?
+            OpenNebula.log_error("Cannot load max number of nbd devices\n#{e}")
+            return 0
```
Comment (rsmontero) [inline src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb]: Fallback to previous constant NBD_MAX

File: src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb [hunk 2]
```diff
@@ -67,6 +65,21 @@ def do_unmap(device, _one_vm, _disk, _directory)
...
+        end
+
+        param = mods_conf[/(nbds_max=)(\d+)/]
+        param.split('=').last.to_i
```
Comment (rsmontero) [inline src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb]: Make use of pattern matching to get block, i.e. groups (\d+) are returned by the matched object no need to split.last

File: src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb [hunk 3]
```diff
@@ -67,6 +65,21 @@ def do_unmap(device, _one_vm, _disk, _directory)
 
     private
 
+    # Detects Max number of block devices
+    def load_nbds_max
```
Comment (rsmontero) [inline src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb]: better without load

**Pull request merged**